android/src/main/java/com/swmansion/reanimated/nodes/Node.java

```diff
@@ -76,54 +76,105 @@ public void removeChild(Node child) {
     }
   }
 
-  protected void markUpdated() {
+  public void onDrop() {
+    // no-op
+  }
+
+  public EvalContext contextForUpdatingChildren(EvalContext context, Node lastVisited) {
+    return context;
+  }
+
+  protected void markUpdated(EvalContext context) {
+    // It's unnecessary to mark as updated nodes in non-global context, because each change
+    // of node -in  non-global context has to be triggered in
+    // global context (e.g. by change of value or clock)
+    if (context != mNodesManager.globalEvalContext) {
```

Comment:
Can you verify this somehow? E.g. if spring is implemented using procnode and I make two springs then only one will be hooked to an active clock, does it mean the same number of nodes will be visited during evaluation as is we didn't do that with proc nodes?

Reply:
Yes, of course. I mean that every value nodes are stored in global context. And it's not possible to mark updated any non-value node ad updated without updating value node